src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -47,6 +47,59 @@ function CompleteBipartiteGraph(n1::T, n2::T) where {T <: Integer}
     return SimpleGraph(ne, fadjlist)
 end
 
+"""
+    CompleteMultipartiteGraph(partitions)
+
+Create an undirected [complete bipartite graph](https://en.wikipedia.org/wiki/Complete_bipartite_graph)
+with `sum(partitions)` vertices and `prod(partitions)` edges. A partition with `0` vertices is skipped.
+"""
+function CompleteMultipartiteGraph(partitions::AbstractVector{T}) where {T <: Integer}
+    any(x -> x < 0, partitions) && return SimpleGraph{T}(0)
+    length(partitions) == 1 && return SimpleGraph{T}(partitions[1])
+    length(partitions) == 2 && return CompleteBipartiteGraph(partitions[1], partitions[2])
+
+    n = sum(partitions)
+
+    ne = 0
+    for p in partitions # type stability fails if we use sum and a generator here
+        ne += p*(Int(n)-p) # overflow if we don't convert to Int
+    end
+    ne = div(ne, 2)
+
+    fadjlist = Vector{Vector{T}}(undef, n)
+    cur = 1
+    for p in partitions
+        currange = cur:(cur+p-1) # all vertices in the current partition
+        lowerrange = 1:(cur-1)   # all vertices lower than the current partition
+        upperrange = (cur+p):n   # all vertices higher than the current partition
+        @inbounds @simd for u in currange
+            fadjlist[u] = vcat(Vector{T}(lowerrange), Vector{T}(upperrange))
```

Comment:
If you rewrite 
`fadjlist[u] = vcat(Vector{T}(lowerrange), Vector{T}(upperrange))`
as something like
```
fadjlist[u] = Vector{T}(undef, length(lowerrange) + length(uperrange))
fadjlist[1:length(lowerrange)] = lowerrange
fadjlist[length(lowerrange)+1:end] = uperrange
```
then you will not have to allocate some memory that will be freed right afterwards.